src/services/formatting/formatting.ts

```diff
@@ -387,7 +389,10 @@ namespace ts.formatting {
 
             let indentation = inheritedIndentation;
             if (indentation === Constants.Unknown) {
-                if (isSomeBlock(node.kind)) {
+                if (isIndentPreventedChildNode(parent.kind, node.kind)) {

```

Comment:
i would inline the body of isIndentPreventedChildNode, 
```ts
if (parent.kind === SyntaxKind.JsxElement && child.kind === SyntaxKind.JsxClosingElement)
```
it is more readable this way.